apps/champions/lib/champions/users.ex

```diff
@@ -195,4 +206,37 @@ defmodule Champions.Users do
       {:error, _, reason, _} -> {:error, reason}
     end
   end
+
+  @doc """
+  Level up the Kaline Tree of a user.
+  """
+  def level_up_kaline_tree(user_id) do
+    with {:user, {:ok, user}} <- {:user, Users.get_user(user_id)},
+         level_up_costs = calculate_costs_to_level_up_kaline_tree(user),
+         {:can_afford, true} <-
+           {:can_afford,
+            Enum.all?(level_up_costs, fn %CurrencyCost{currency_id: level_up_currency_id, amount: level_up_cost} ->
```

Comment:
We already have a function to iterate through CurrencyCosts: https://github.com/lambdaclass/mirra_backend/blob/671689978f33f6819b7ff7e627ba34bf4722da54/apps/game_backend/lib/game_backend/users/currencies.ex#L105